NobBS-Experiments/Experiment1.R

```diff
@@ -0,0 +1,44 @@
+library(jsonlite)
+library(magrittr)
+library(dplyr)
+library(fGarch)
+library(NobBS)
+
+# Get raw patient data
+raw_patient_df <- data.frame(fromJSON("https://api.covid19india.org/raw_data.json"))
+
+# Get names of all districts with patients
+district_wise_data <- fromJSON("https://api.covid19india.org/state_district_wise.json")
+districts_list = list()
+for(p in district_wise_data) { districts_list[names(p$districtData)] <- FALSE}
+districts_list[["Unknown"]] <- NULL
+districts_df = data.frame(names(districts_list))
+colnames(districts_df) <- c("district")
+
+# cleaning up patient data to remove tourist related data and retaining only the columns needed
+patient_clean_df = merge(x=districts_df, y=raw_patient_df, by.x='district', by.y='raw_data.detecteddistrict')
+patient_data <- patient_clean_df %>% select(district, raw_data.dateannounced, raw_data.gender)
+patient_data_reliable <- patient_data %>% filter(raw_data.dateannounced >= "01/03/2020")
+
+#### Calculation of onset date based on announcement date
+num_patients <- nrow(patient_data_reliable)
+# Using Skewed Gaussian with a mean of 14 days and standard deviation 1 to get the incubation time
+incubation_times <- floor(rsnorm(num_patients, mean=14, sd=1, xi=-10))
```

Comment:
```suggestion
incubation_times <- floor(rsnorm(num_patients, mean=9, sd=5, xi=-10))
```
Mean of 9 days, sd of 5 and what does `xi` mean here?

Reply:
My idea here was to use a Skewed Gaussian instead of Gaussian itself. My intuition came from the fact that in most cases, it takes 14 days for incubation (I could be wrong here). 'xi' is the skewness and making it -ve has made left skewed (meaning we expect more number of people to have an incubation period of 14. Attached is how the distribution looks for incubation periods I have used. Let me know if this makes sense, else, we can stick to normal curve with mean 9 and standard deviation 5
![Incubation time graph](https://user-images.githubusercontent.com/29398129/80300869-da582580-87bd-11ea-8726-7d39929c29d2.png)

Reply:
While the skewed Gaussian assumption makes sense, the 14 days number is debatable. 
Here is my recommendation: Mean of 9 days with `sd=5`, with a skew towards 14 days. 

### Why this recommendation?

Incubation Period (`IncubPeriod`) has a _best estimate_ of mean is 5 days with range from 3.8 to 9 days.

NobBS is looking for _disease onset time_ IIRC, which is from incubation to symptoms. There is an unknown time from incubation to symptoms onset, we'll assume that to be zero -- since incubation time is often measured by first symptoms.

This is where things get interesting. The definition of symptoms varies by country to country. In France, loss of smell and taste is considered a symptom - while in S. Korea, even mild fever alone is a symptom. 

What India would consider as symptom, would be typically _severe infection_ symptoms in Wuhan/S. Korea. This variable `DurMildInf`'s mean/median best estimate is 6 days with a range of 5 to 12 days.

Based on above, I'm simply making a guesstimate of `sd` and I'm okay with any value larger than 3 days. I'll let you make an informed guess for `xi`. Consider checking out the Korea CDC papers mention in the link below.

Source for numbers above: https://bit.ly/COVID19_Params

Reply:
Makes sense. Changed the mean and standard deviation to 9 and 4. Made adjustments for xi and for removing negative values that have come up in the distribution. The distribution looks as below now:
![Incubation time graph](https://user-images.githubusercontent.com/29398129/80304132-d6ce9980-87d1-11ea-87fa-53f332db6e37.png)